features/semaphore/semaphoreSelector.js

```diff
@@ -25,41 +25,47 @@ export const getInput = createSelector(
   }
 )
 
-const getHorizontalFlip = createSelector([getInput], (input) => {
-  return input.map((item) => {
-    return [...item].map((button) => {
-      return {
-        1: 1,
-        2: 8,
-        3: 7,
-        4: 6,
-        5: 5,
-        6: 4,
-        7: 3,
-        8: 2,
-      }[button]
-    })
-  })
-})
-
 export const getAllResults = createSelector(
-  [getInput, getHorizontalFlip, getVariantId],
-  (input, horizontalFlip, variantId) => {
-    const allVariants = [
+  [getInput, getVariantId],
+  (input, variantId) => {
+    const partLength = 8
+    const baseOrders = [...Array(partLength).keys()].map((item) => item + 1)
+    const variantOrders = variantPermutations(baseOrders, false).slice(1)
```

Comment:
I think this could be written as 

```suggestion
    const variantOrders = variantPermutations(baseOrders)
```


Anyway, do it as you want.